resolver.go

```diff
@@ -200,6 +200,9 @@ func (r *resolver) ServeDNS(w dns.ResponseWriter, query *dns.Msg) {
 		err  error
 	)
 
+	if len(query.Question) == 0 {
```

Comment:
query can't be nil; miekg/dns library calls ServeDNS with a new dns.Msg. But doesn't hurt to add defensive check..